### PR TITLE
Holaplex indexer

### PR DESCRIPTION
**Pull request by dandlezzz**
Adding the Holaplex indexer for fetching nfts

### CODE REVIEW

File: utils/tokens.tsx [hunk 1]
```diff
@@ -370,37 +371,62 @@ export const deserializeMint = (data: Buffer) => {
...
-    return arr
+      `,
+      variables: {
+        owners: [owner.toBase58()],
+      },
+    }),
+  })
+
+  let body = await result.json()
+  console.log({ body })
```
Comment (kespinola) [inline utils/tokens.tsx:418]: drop console log before ship.

File: utils/tokens.tsx [hunk 2]
```diff
@@ -370,37 +371,62 @@ export const deserializeMint = (data: Buffer) => {
...
+  return body.data
+}
+
+export const getNfts = async (
+  connection: Connection,
+  ownerPk: PublicKey
+): Promise<NFTWithMeta[]> => {
+  try {
+    const data = await fetchNftsFromHolaplexIndexer(ownerPk)
+    console.log('here are the nfts', data.nfts)
```
Comment (kespinola) [inline utils/tokens.tsx:428]: drop console.log before ship.

File: hooks/useVotingPlugins.ts [hunk 3]
```diff
@@ -108,20 +108,8 @@ export function useVotingPlugins() {
...
-      const nftsWithMeta = votingNfts.map((x) => {
-        const nft = nfts.find(
-          (nft) => nft.tokenAddress === x.tokenAddress.toBase58()
-        )
-        return {
-          ...nft!,
-          metadata: x.metadata,
-        }
-      })
+      const votingNfts = nfts.filter((x) => getIsFromCollection(x))
```
Comment (kespinola) [inline hooks/useVotingPlugins.ts:111]:
```suggestion
      const votingNfts = nfts.filter(getIsFromCollection)
```

File: utils/tokens.tsx [hunk 4]
```diff
@@ -370,37 +371,62 @@ export const deserializeMint = (data: Buffer) => {
...
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({
+      query: `
+        query nfts($owners: [PublicKey!]) {
+            nfts(
+              owners: $owners,
+               limit: 10000, offset: 0) {
```
[the remark this answers is not in the document]
Comment (dandlezzz) [reply utils/tokens.tsx:382]: Good question. The largest example I  was given was 300, so I'm hopeful this will work for nearly all use cases.

**Review by nramadas — CHANGES_REQUESTED** [review]
Hey @dandlezzz ! Noticed a small issue. When I visit a DAO, it show all of my NFTs, not just the ones relevant to the DAO. For example on https://governance-907qtch70-blockworks-foundation.vercel.app/dao/DTP, i see:
image

But on prod (https://realms.today/dao/DTP), I see:
image

Maybe the nft list isn't being fully filtered?

**Review by nramadas — APPROVED** [review]

File: NftVotePlugin/store/nftPluginStore.ts [hunk 5]
```diff
@@ -39,14 +39,11 @@ const useNftPluginStore = create<nftPluginStore>((set, _get) => ({
     })
   },
   setVotingNfts: (nfts, votingClient, nftMintRegistrar) => {
-    const filteredNfts = nfts.filter(
-      (x) => x.token.account.amount.cmpn(1) === 0
-    )
-    votingClient._setCurrentVoterNfts(filteredNfts)
+    votingClient._setCurrentVoterNfts(nfts)
```
Comment (abrzezinski94) [inline NftVotePlugin/store/nftPluginStore.ts:42]: We can't remove the filter becuase only nfts with amount === 1 can vote.

**Review by nramadas — CHANGES_REQUESTED** [review]
Hey @dandlezzz , I think i might have spotted another issue.

In this DAO, on prod, there seems to be an NFT in its treasury:
https://realms.today/dao/MNGO

image

But, in this branch, that NFT no longer shows up:
https://governance-6sstw70o3-blockworks-foundation.vercel.app/dao/MNGO

image

Would you mind taking a look?

**Comment by dandlezzz** [comment]
Looking into this issue.

**Comment by dandlezzz** [comment]
> Hey @dandlezzz , I think i might have spotted another issue.
> 
> In this DAO, on prod, there seems to be an NFT in its treasury: https://realms.today/dao/MNGO
> 
> image
> 
> But, in this branch, that NFT no longer shows up: https://governance-6sstw70o3-blockworks-foundation.vercel.app/dao/MNGO
> 
> image
> 
> Would you mind taking a look?

Looking into this.

**Review by nramadas — APPROVED** [review]

**Pull request merged**